src/cli/main.cpp

```diff
@@ -31,174 +31,289 @@ bool cancel()
     return false;
 }
 
+std::string shorter(const char* shorter)
+{
+    return std::string(shorter) + ",";
+}
+
+auto string_with_default(const char* defs) -> decltype(cxxopts::value<std::string>()->default_value(""))
+{
+    return cxxopts::value<std::string>()->default_value(defs);
+}
+
+auto double_with_default(const char* defs) -> decltype(cxxopts::value<double>()->default_value(""))
+{
+    return cxxopts::value<double>()->default_value(defs);
+}
+
+auto int_with_default(const char* defs) -> decltype(cxxopts::value<int>()->default_value(""))
```

Comment:
@iglesias any idea how to avoid code duplication here? It seems template aliases are only supported for types but I wanted to avoid introducing classes.

Reply:
Yeah, it's in #87. I was doubting if the return type deduction would be problematic with compiler version but locally with gcc 13.2 it worked just fine without tweaking any options. I don't know in the CI yet. Can you take a look and try if it works well on your end too?

Reply:
Would it make sense trying to simplify more passing int, double or std::string (instead of the literals)? Deducing the type and then no need to pass it, e.g. toward `with_default(2)` instead of `int_with_default("2")` or `with_default<int>("2")`.